### DISCUSSION TITLE
Upload custom scripts via git-repo in folder

### DISCUSSION THREAD

**Discussion by lp-2** [netbox-community/netbox#12663] [Help Wanted!]
Unfortunately, I cannot find a way in the documentation to load custom scripts by moving them to the scripts folder.

the following situation, the custom scripts are maintained via git-repo, but since version 3.5.1 they have to be uploaded manually via web UI, as soon as these scripts are added directly to the folder, netbox ignores or does not load them.

What would be the procedure for such a case? 
because this is not described in the documentation (https://docs.netbox.dev/en/stable/customization/custom-scripts/).

**Comment by zmaster7** [comment]
I believe you would need to use the new DataSource option: [https://docs.netbox.dev/en/feature/models/core/datasource/](https://docs.netbox.dev/en/feature/models/core/datasource/)

You can either point a data source to a local location, or directly to the git-repo. Then, add a script using the DataSource/File and choose "Auto sync enabled."

![image](https://github.com/netbox-community/netbox/assets/46574676/8a020a94-ce09-4938-bfd5-1383e4ffa148)

Each script would have to be manually added once using this method though, as I don't believe there is a method to add them via API.

**Reply by zmaster7** [reply]
Looks like #12624 was opened to address this as well.
